tests/test_tgis_backend.py

```diff
@@ -681,6 +682,90 @@ def test_tgis_backend_config_load_prompt_artifacts():
                 tgis_be.load_prompt_artifacts("buz", prompt_id1, source_files[0])
 
 
+@pytest.mark.parametrize(
+    argnames=["model_id", "conn_cfg", "fill", "expected_conn_cfg"],
+    argvalues=[
+        (
+            "model1",
+            None,
+            False,
+            {
+                "hostname": "localhost:1234",
+                "model_id": "model1",
+                "lb_policy": "abc",
+            },
+        ),
+        (
+            "model1",
+            None,
+            True,
+            {
+                "hostname": "localhost:1234",
+                "model_id": "model1",
+                "lb_policy": "abc",
+            },
+        ),
+        (
+            "model1",
+            {"hostname": "myhost"},
+            False,
+            {"hostname": "myhost", "model_id": "model1"},
+        ),
+        (
+            "model1",
+            {"hostname": "myhost"},
+            True,
+            {"hostname": "myhost", "model_id": "model1", "lb_policy": "abc"},
+        ),
+    ],
+)
+def test_tgis_backend_register_model_connection(
+    model_id: str,
+    conn_cfg: Optional[dict],
+    fill: bool,
+    expected_conn_cfg: Dict[str, Any],
+):
+    """Test that register_model_connection correctly adds a TGISConnection to the _model_connections dictionary"""
+    tgis_be = TGISBackend(
+        {
+            "connection": {"hostname": "localhost:1234", "grpc_lb_policy_name": "abc"},
+            "remote_models": {},
+        }
+    )
+
+    # Assert new model is not in backend
+    assert model_id not in tgis_be._remote_models_cfg
+    assert model_id not in tgis_be._model_connections
+
+    # Register model
+    tgis_be.register_model_connection(model_id, conn_cfg, fill_with_defaults=fill)
+    assert model_id in tgis_be._remote_models_cfg
+    assert model_id in tgis_be._model_connections
+    assert isinstance(tgis_be._model_connections[model_id], TGISConnection)
+    assert {
+        k: v
+        for k, v in asdict(tgis_be._model_connections[model_id]).items()
+        if v is not None
+    } == expected_conn_cfg
+
+    # Re-register -> no change to existing model
+    tgis_be.register_model_connection(model_id, {"hostname": "{model_id}.mycluster"})
+    assert {
+        k: v
+        for k, v in asdict(tgis_be._model_connections[model_id]).items()
+        if v is not None
+    } == expected_conn_cfg
+
+    # Confirm get_connection works
+    conn = tgis_be.get_connection(model_id, create=False)
+    assert isinstance(conn, TGISConnection)
+    assert {
+        k: v
+        for k, v in asdict(tgis_be._model_connections[model_id]).items()
+        if v is not None
+    } == expected_conn_cfg
+
```

Comment:
Per the comment about not mutating the base config, I think we need one more clause here to test another non-override template `get_connection` to ensure that the base `hostname` has not been changed.